Import error

Hi

I'm trying to import an OVA file but get the following error from logs:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12b5746]

goroutine 1 [running]:
github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_ovf_import/ovf_utils.GetOSId(0xc000130930, 0xc000166aa0, 0xc000392140, 0x41, 0xc000130930)
	/workspace/cli_tools/gce_ovf_import/ovf_utils/ovf_utils.go:261 +0xe6
github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_ovf_import/ovf_importer.(*OVFImporter).setUpImportWorkflow(0xc0001d6000, 0x193ffbe, 0x1d, 0x5)
	/workspace/cli_tools/gce_ovf_import/ovf_importer/ovf_importer.go:431 +0x395
github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_ovf_import/ovf_importer.(*OVFImporter).Import(0xc0001d6000, 0xc0001d6000, 0x0, 0x0)
	/workspace/cli_tools/gce_ovf_import/ovf_importer/ovf_importer.go:474 +0x6f
main.runImport(0x0, 0x0, 0x0, 0x0)
	/workspace/cli_tools/gce_ovf_import/main.go:114 +0x8b
github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/logging/service.(*Logger).runWithServerLogging(0xc0000e6fc0, 0x197b5c0, 0xc000549250, 0x0, 0xc000118900, 0x0)
	/workspace/cli_tools/common/utils/logging/service/logger.go:207 +0xa3
github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/logging/service.RunWithServerLogging(0x191ea0b, 0xe, 0x0, 0x0, 0xc000118900, 0x0, 0xc000549250, 0x197b5c0, 0x1, 0xc000076058)
	/workspace/cli_tools/common/utils/logging/service/logger.go:244 +0x86
main.main()
	/workspace/cli_tools/gce_ovf_import/main.go:133 +0xe4
ERROR
ERROR: build step 0 "gcr.io/compute-image-tools/gce_ovf_import:release" failed: step exited with non-zero status: 2
```
The VM specifications are as follows:
```
Virtualization: oracle
Operating System: Debian GNU/Linux 9 (stretch)
Kernel: Linux 4.9.0-8-amd64
Architecture: x86-64
```
I was hoping to understand what is causing the seg-fault.

Thanks in advance!

--- a/cli_tools/gce_ovf_import/ovf_utils/ovf_utils.go
+++ b/cli_tools/gce_ovf_import/ovf_utils/ovf_utils.go
@@ -256,6 +256,9 @@ func GetOSId(ovfDescriptor *ovf.Envelope) (string, error) {
 		len(ovfDescriptor.VirtualSystem.OperatingSystem) == 0 {
 		return "", daisy.Errf("OperatingSystemSection must be defined to retrieve OS info")
 	}
+	if ovfDescriptor.VirtualSystem.OperatingSystem[0].OSType == nil {
+		return "", daisy.Errf("OperatingSystemSection.OSType must be defined to retrieve OS info")
+	}
 	var osID string
 	var validOSType bool
 	osType := *ovfDescriptor.VirtualSystem.OperatingSystem[0].OSType

--- a/cli_tools/gce_ovf_import/ovf_utils/ovf_utils_test.go
+++ b/cli_tools/gce_ovf_import/ovf_utils/ovf_utils_test.go
@@ -649,12 +649,25 @@ func TestGetOSIdNonDeterministicMultiOption(t *testing.T) {
 		err.Error())
 }
 
+func TestGetOSIdNilOSIdInDescriptor(t *testing.T) {
+	osID, err := GetOSId(createOVFDescriptorWithOSTypeAsReference(nil))
+	assert.Equal(t, "", osID)
+	assert.NotNil(t, err)
+	assert.Equal(t,
+		"OperatingSystemSection.OSType must be defined to retrieve OS info",
+		err.Error())
+}
+
 func createOVFDescriptorWithOSType(osType string) *ovf.Envelope {
+	return createOVFDescriptorWithOSTypeAsReference(&osType)
+}
+
+func createOVFDescriptorWithOSTypeAsReference(osType *string) *ovf.Envelope {
 	return &ovf.Envelope{
 		VirtualSystem: &ovf.VirtualSystem{
 			OperatingSystem: []ovf.OperatingSystemSection{
 				{
-					OSType: &osType,
+					OSType: osType,
 				},
 			},
 		},

